ras_party/views/party_view.py

```diff
@@ -79,6 +79,19 @@ def get_respondent_by_email(email):
     return controller.get_respondent_by_email(email)
 
 
+@party_view.route('/respondents/email', methods=['PUT'])
+@auth.login_required
+@log_route
+def change_respondent_email():
+    request_data = request.json
+    email_address = request_data.get('email_address')
+    new_email_address = request_data.get('new_email_address')
+    if not email_address or not new_email_address:
+        return make_response(jsonify({'errors': 'No email provided'}), 400)
+
+    return controller.change_respondent_email(email_address, new_email_address)
```

Comment:
Can change-respondent_email handle a None for email_address? Also session isn't passed here, maybe set a default for session.

Reply:
If the email_address is `None` then the 400 error on line 90 will trigger
And the session actually get's passed to `controller.change_respondent_email` by its decorator, `with_db_session`

Reply:
Actually I think you mean how the function itself would deal with a None for email_address, I'm not 100% sure whether the database query on line 642(!) of controller.py would fail or return None

Reply:
I'd be inclined to pass the entire payload through to the controller function and add the defensive code there. For example, see respondents_post for validating required fields.